guides/common/assembly_registering-hosts.adoc

```diff
@@ -50,3 +50,13 @@ include::modules/proc_installing-tracer.adoc[leveloffset=+1]
 include::modules/proc_installing-and-configuring-puppet-agent-during-host-registration.adoc[leveloffset=+1]
 
 include::modules/proc_installing-and-configuring-puppet-agent-manually.adoc[leveloffset=+1]
+
+include::modules/con_using-custom-ssl-certificate-for-hosts.adoc[leveloffset=+1]
+
+ifdef::context[:parent-context: {context}]
+:context: {project-context}
+include::modules/proc_deploying-a-custom-ssl-certificate-to-hosts.adoc[leveloffset=+2]
+ifdef::parent-context[:context: {parent-context}]
+ifndef::parent-context[:!context:]
```

Comment:
What's the reasoning behind this? Would this work too:

```suggestion
include::modules/proc_deploying-a-custom-ssl-certificate-to-hosts.adoc[leveloffset=+2]
```

Reply:
`context` rendered `managing-hosts` when we added this file to the assembly, which is incorrect. `project-context` will render the desired value.

Reply:
but "registering hosts" is part of the "Managing Hosts" guide: https://docs.theforeman.org/nightly/Managing_Hosts/index-katello.html#registering_hosts_to_server_managing-hosts

Reply:
This line https://github.com/theforeman/foreman-documentation/pull/2886/files#diff-c71be99ed7ce20d915d69deac63ffab56c0e7bd8f1fe6ceda64b94cf05bbe042 rendered `managed-hosts.example.com`. So, to have the correct value, we changed the context exclusive to the file `proc_deploying-a-custom-ssl-certificate-to-hosts.adoc` in this assembly.

Reply:
Using "{context}.example.com" is not really a good idea. Instead, we generally use {foreman-example-com} or {smartproxy-example-com}. Which line is it? The link does not work for me/I cannot find this line yet.

Reply:
Line 12 of `proc_deploying-a-custom-ssl-certificate-to-hosts.adoc`.

I guess the' context' is used to satisfy `project` and `smartproxy` both.

Reply:
I belive this is a similar issue to https://github.com/theforeman/foreman-documentation/pull/2832#discussion_r1530291462

I think it's a bad idea to mimic the "foreman.example.com" value of "{foreman-example-com}". Instead, we should wrap the module with a specific attribute.

Reply:
Addressed :+1:
I am flexible to naming convention of attributes.-